design/crud-api/db/repository/postgres/data_handler.go

```diff
@@ -111,39 +115,545 @@ func logSchemaInfo(schemaInfo *schema.SchemaInfo) {
+	var attributeID int
+	err = client.DB().QueryRowContext(ctx,
+		`INSERT INTO entity_attributes (entity_id, attribute_name, table_name)
+		VALUES ($1, $2, $3)
+		ON CONFLICT (entity_id, attribute_name) DO UPDATE
+		SET table_name = EXCLUDED.table_name
+		RETURNING id`,
+		entityID, attrName, tableName).Scan(&attributeID)
+	if err != nil {
+		return fmt.Errorf("error creating entity attribute record: %v", err)
+	}
+
+	// Extract data from the TimeBasedValue
+	var tabularStruct structpb.Struct
+	if err := value.Value.UnmarshalTo(&tabularStruct); err != nil {
+		return fmt.Errorf("error unmarshaling tabular data: %v", err)
+	}
+
+	// Extract columns and rows
+	columnsValue := tabularStruct.Fields["columns"].GetListValue()
+	rowsValue := tabularStruct.Fields["rows"].GetListValue()
+
+	if columnsValue == nil || rowsValue == nil {
+		return fmt.Errorf("invalid tabular data format")
+	}
+
+	// Convert columns to string slice
+	columnNames := make([]string, len(columnsValue.Values))
+	for i, col := range columnsValue.Values {
+		columnNames[i] = sanitizeIdentifier(col.GetStringValue())
+	}
+
+	// Convert rows to [][]interface{}
+	rows := make([][]interface{}, len(rowsValue.Values))
+	for i, row := range rowsValue.Values {
+		rowList := row.GetListValue()
+		if rowList == nil {
+			return fmt.Errorf("invalid row format at index %d", i)
+		}
+
+		rows[i] = make([]interface{}, len(rowList.Values))
+		for j, cell := range rowList.Values {
+			switch cell.Kind.(type) {
+			case *structpb.Value_StringValue:
+				rows[i][j] = cell.GetStringValue()
+			case *structpb.Value_NumberValue:
+				rows[i][j] = cell.GetNumberValue()
+			case *structpb.Value_BoolValue:
+				rows[i][j] = cell.GetBoolValue()
+			default:
+				rows[i][j] = cell.GetStringValue()
+			}
+		}
+	}
+
+	// Insert the data
+	if err := client.InsertTabularData(ctx, tableName, attributeID, columnNames, rows); err != nil {
+		return fmt.Errorf("error inserting tabular data: %v", err)
+	}
+
+	return nil
+}
+
+// schemaToColumns converts a schema to database columns
+func schemaToColumns(schemaInfo *schema.SchemaInfo) []Column {
+	var columns []Column
+
+	for fieldName, field := range schemaInfo.Fields {
+		var colType string
+		switch field.TypeInfo.Type {
+		case typeinference.IntType:
+			colType = "INTEGER"
+		case typeinference.FloatType:
+			colType = "DOUBLE PRECISION"
+		case typeinference.StringType:
+			colType = "TEXT"
+		case typeinference.BoolType:
+			colType = "BOOLEAN"
+		case typeinference.DateType:
+			colType = "DATE"
+		case typeinference.DateTimeType:
+			colType = "TIMESTAMP WITH TIME ZONE"
+		default:
+			colType = "TEXT"
+		}
+
+		if field.TypeInfo.IsNullable {
+			colType += " NULL"
+		} else {
+			colType += " NOT NULL"
+		}
+
+		columns = append(columns, Column{
+			Name: sanitizeIdentifier(fieldName),
+			Type: colType,
+		})
+	}
+
+	return columns
+}
+
+// sanitizeIdentifier makes a string safe for use as a PostgreSQL identifier
+func sanitizeIdentifier(s string) string {
```

Comment:
this is cool. Let's create an issue so that later we use something like a sanitize.go kinda file to keep these sanitizers. We do this such that this class data_handler.go only do the data handling part, any other utils or helpers should live somewhere else in a re-usable way. 

Also paste the issue here IMPROVEME: <issue_url> just above the code.